File: entities/document.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type DocumentType string
type OCRStatus string

const (
	DocumentTypeRegistrationCertificate DocumentType = "REGISTRATION_CERTIFICATE"
	DocumentTypeMBBSCertificate         DocumentType = "MBBS_CERTIFICATE"
	DocumentTypeMDCertificate           DocumentType = "MD_CERTIFICATE"
	DocumentTypeAadhaar                 DocumentType = "AADHAAR"
	DocumentTypePAN                     DocumentType = "PAN"
	DocumentTypePassport                DocumentType = "PASSPORT"

	OCRStatusPending    OCRStatus = "PENDING"
	OCRStatusProcessing OCRStatus = "PROCESSING"
	OCRStatusCompleted  OCRStatus = "COMPLETED"
	OCRStatusFailed     OCRStatus = "FAILED"
)

type DoctorDocument struct {
	DocumentID       uuid.UUID      `gorm:"primaryKey" json:"document_id"`
	DoctorID         uuid.UUID      `gorm:"not null;index" json:"doctor_id"`
	DocumentType     DocumentType   `gorm:"size:50;not null;index" json:"document_type"`
	FileURL          string         `gorm:"type:text;not null" json:"file_url"`
	OriginalFilename string         `gorm:"size:255;not null" json:"original_filename"`
	MIMEType         string         `gorm:"size:100;not null" json:"mime_type"`
	FileSize         int64          `gorm:"not null" json:"file_size"`
	FileHash         string         `gorm:"size:64;not null;index" json:"file_hash"`
	ResolutionWidth  *int           `json:"resolution_width,omitempty"`
	ResolutionHeight *int           `json:"resolution_height,omitempty"`
	Version          int            `gorm:"default:1;not null" json:"version"`
	IsLatest         bool           `gorm:"default:true;not null" json:"is_latest"`
	OCRStatus        OCRStatus      `gorm:"size:50;default:'PENDING';not null" json:"ocr_status"`
	UploadedAt       time.Time      `gorm:"autoCreateTime" json:"uploaded_at"`
	UpdatedAt        time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt        gorm.DeletedAt `gorm:"index" json:"-"`

	// Relations
	OCRResults []DocumentOCRResult `gorm:"foreignKey:DocumentID;constraint:OnDelete:CASCADE" json:"ocr_results,omitempty"`
}

func (d *DoctorDocument) BeforeCreate(tx *gorm.DB) (err error) {
	if d.DocumentID == uuid.Nil {
		d.DocumentID = uuid.New()
	}
	return nil
}

File: python_backend/entities/document.py
import enum
import uuid
from sqlalchemy import Column, String, Integer, BigInteger, Boolean, DateTime, Text, Enum, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from database import Base

class DocumentType(str, enum.Enum):
    REGISTRATION_CERTIFICATE = "REGISTRATION_CERTIFICATE"
    MBBS_CERTIFICATE = "MBBS_CERTIFICATE"
    MD_CERTIFICATE = "MD_CERTIFICATE"
    MEDICAL_DEGREE_CERTIFICATE = "MEDICAL_DEGREE_CERTIFICATE"
    PG_CERTIFICATE = "PG_CERTIFICATE"
    AADHAAR = "AADHAAR"
    PAN = "PAN"
    PASSPORT = "PASSPORT"

class OCRStatus(str, enum.Enum):
    PENDING = "PENDING"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"

class DoctorDocument(Base):
    __tablename__ = "doctor_documents"

    document_id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    document_type = Column(Enum(DocumentType, name="document_type_enum", native_enum=False), nullable=False, index=True)
    file_url = Column(Text, nullable=False)
    original_filename = Column(String(255), nullable=False)
    mime_type = Column(String(100), nullable=False)
    file_size = Column(BigInteger, nullable=False)
    file_hash = Column(String(64), nullable=False, index=True)
    resolution_width = Column(Integer, nullable=True)
    resolution_height = Column(Integer, nullable=True)
    version = Column(Integer, default=1, nullable=False)
    is_latest = Column(Boolean, default=True, nullable=False)
    ocr_status = Column(Enum(OCRStatus, name="ocr_status_enum", native_enum=False), default=OCRStatus.PENDING, nullable=False)
    uploaded_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), index=True, nullable=True)

    # Relations
    ocr_results = relationship("DocumentOCRResult", backref="document", cascade="all, delete")
